staging/src/k8s.io/apimachinery/pkg/api/resource/quantity_test.go

```diff
@@ -215,6 +215,39 @@ func TestParseQuantityString(t *testing.T) {
 	}
 }
 
+func TestQuantityToInt64(t *testing.T) {
+	table := []struct {
+		name       string
+		v          string
+		isOverflow bool
+	}{
+		{"decimalSI to int64 not overflow",
+			"1000000000G",
+			true,
+		},
+		{"binarySI to int64 overflow",
+			"100000000Gi",
+			true,
+		},
+		{"decimalSI to int64 overflow",
+			"9223372036854775807",
+			false,
+		},
+		{"binarySI to int64 overflow",
+			"1000000000Gi",
```

Comment:
Sorry for being pedantic here-- technically we are not "overflowing". 1000000000Gi < max int64. What happened was that we cannot easily determine if that's the case with simple string comparison, therefore we store 1000000000Gi in inf.Dec form. 
- If an inf.Dec <= max int64, it's value will be the same
- If an inf.Dec > max int64, it will be "overflowing" and become [maxAllowed](https://github.com/kubernetes/kubernetes/blob/813671d1a0bb90a48c220916b4d2e9fc8e8c9b9a/staging/src/k8s.io/apimachinery/pkg/api/resource/math.go#L44) (max int64)

Reply:
Yes, I personally believe it's pretty good. Storing binary data in decimal can improve the performance of judging "overflowing", even if there is a loss in accuracy. As long as it is correctly judged that data larger than MaxInt64 is “overflowing”, most users will accept it.